sandpack-react/src/templates/react-typescript.ts

```diff
@@ -0,0 +1,86 @@
+import type { SandboxTemplate } from "../types";
+
+export const REACT_TYPESCRIPT_TEMPLATE: SandboxTemplate = {
+  files: {
+    "tsconfig.json": {
+      code: `{
+    "include": [
+        "./**/*"
+    ],
+    "compilerOptions": {
+        "strict": true,
+        "esModuleInterop": true,
+        "lib": [
+            "dom",
+            "es2015"
+        ],
+        "jsx": "react-jsx"
+    }
+}`,
+    },
+    "/App.tsx": {
+      code: `export default function App() {
+  return <h1>Hello World</h1>
+}
+`,
+    },
+    "/index.tsx": {
+      code: `import React, { StrictMode } from "react";
+import ReactDOM from "react-dom";
+import "./styles.css";
+
+import App from "./App";
+
+const rootElement = document.getElementById("root");
+ReactDOM.render(
+  <StrictMode>
+    <App />
+  </StrictMode>,
+  rootElement
+);`,
+    },
+    "/styles.css": {
+      code: `body {
+  font-family: sans-serif;
+  -webkit-font-smoothing: auto;
+  -moz-font-smoothing: auto;
+  -moz-osx-font-smoothing: grayscale;
+  font-smoothing: auto;
+  text-rendering: optimizeLegibility;
+  font-smooth: always;
+  -webkit-tap-highlight-color: transparent;
+  -webkit-touch-callout: none;
+}
+
+h1 {
+  font-size: 1.5rem;
+}`,
+    },
+    "/public/index.html": {
+      code: `<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <meta charset="UTF-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <title>Document</title>
+  </head>
+  <body>
+    <div id="root"></div>
+  </body>
+</html>`,
+    },
+  },
+  dependencies: {
+    react: "^17.0.0",
+    "react-dom": "^17.0.0",
+    "react-scripts": "^4.0.0",
+  },
+  devDependencies: {
```

Comment:
not 100% sure, but I think create-react-app works with typescript as long as you provide a tsconfig file. @CompuIves I'm curious what's your take on differentiating `dependencies` and `devDependencies` in the sandpack API. Are there any differences in how they are treated in the bundler?

Reply:
I tried to mimic the "React Typescript" template.
IMO this is not only about the React types but also about types for 3rd party libraries which might be included in the sandbox.
If custom `devDependencies` are not yet supported, then maybe a possible workaround for sandpack would be to provide the `@types` as runtime `dependencies`?

Reply:
Hi @alexnm, are there any news about this topic?

Reply:
Heya! Yeah there is a difference, we generally don't install `devDependencies`, because most tools in there were build tools, while sandpack is a build tool, and it increased bandwidth usage quite a bit.

However, there were plans to support `devDependencies` by going through the application code and detecting which dependencies are used, and only using those. But that would've required us to rewrite the parsing logic (only install dependencies after first parse). So that's still on the backlog.

Reply:
But! The general idea is that `devDependencies` should be there, because when you download the repository locally you _need_ to be able to run it as well. So it's good that it's added here.

Reply:
And, `@types/...` is used in CodeSandbox editor to dynamically download type definitions, even if it's defined in `devDependencies`. This is not used in Sandpack though, since it only cares about bundling. I would keep it here!